packages/core/src/textures/resources/CanvasResource.js

```diff
@@ -13,11 +13,14 @@ export default class CanvasResource extends BaseImageResource
      * Used to auto-detect the type of resource.
      *
      * @static
-     * @param {*} source - The source object
+     * @param {HTMLCanvasElement|OffscreenCanvas} source - The source object
      * @return {boolean} `true` if <canvas>
      */
     static test(source)
     {
-        return (source instanceof HTMLCanvasElement);
+        return (source && (
+            source instanceof HTMLCanvasElement
+            || source instanceof global.OffscreenCanvas
```

Comment:
the 'global' namespace may not exist.  Shouldn't it work as just `source instanceof OffscreenCanvas`?

Reply:
> the 'global' namespace may not exist

Ah, yes, I see that `global` didn't make it into the final specs... Do you think `globalThis` or `window` would be a better option?

Reply:
`window` works, we use that in other places